File: Tools/skyConverter/skyConverter/ModelStructure/Vertex.cs
using Newtonsoft.Json.Linq;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace skyConverter.ModelStructure
{
    public class Vertex : ISerializer
    {
        private double[] position;
	    private double[] uv;
	    private double[] normal;
	
	    public Vertex(double[] _position, double[] _uv, double[] _normal){
		    position = _position;
		    uv = _uv;
		    normal = _normal;
	    }

        public JObject serializeObject() 
        {
		    JObject obj = new JObject();
		
		    if(position.Length > 0)
            {
		
			    JArray positionNode = new JArray();
			    positionNode.Add(position[0]);
			    positionNode.Add(position[1]);
			    positionNode.Add(position[2]);
			
			    obj.Add("Position", positionNode);
		
		    }
		
		    if(uv.Length > 0)
            {
			
			    JArray uvNode = new JArray();
			    uvNode.Add(uv[0]);
			    uvNode.Add(uv[1]);
			
			    obj.Add("UV", uvNode);
			
		    }
		
		    if(normal.Length > 0)
            {
			    JArray normalNode = new JArray();
			    normalNode.Add(normal[0]);
			    normalNode.Add(normal[1]);
			    normalNode.Add(normal[2]);
			
			    obj.Add("Normal", normalNode);
			
		    }
		
		
		    if (obj.Count > 0){
			    return obj;
		    }

		    return null;
		
	    }

        public JArray serializeArray() 
        {
		    // TODO Auto-generated method stub
		    return null;
	    }
    }
}

File: Tools/Converter/src/toJSON/Vertex.java
package toJSON;

import org.json.simple.JSONArray;
import org.json.simple.JSONObject;


public class Vertex implements ISerializer {

	private double[] position;
	private double[] uv;
	private double[] normal;
	
	public Vertex(double[] _position, double[] _uv, double[] _normal){
		position = _position;
		uv = _uv;
		normal = _normal;
	}

	@Override
	public JSONObject serializeObject() {
		JSONObject obj = new JSONObject();
		
		if(position.length > 0){
		
			JSONArray positionNode = new JSONArray();
			positionNode.add(position[0]);
			positionNode.add(position[1]);
			positionNode.add(position[2]);
			
			obj.put("Position", positionNode);
		
		}
		
		if(uv.length > 0){
			
			JSONArray uvNode = new JSONArray();
			uvNode.add(uv[0]);
			uvNode.add(uv[1]);
			
			obj.put("UV", uvNode);
			
		}
		
		if(normal.length > 0){
			JSONArray normalNode = new JSONArray();
			normalNode.add(normal[0]);
			normalNode.add(normal[1]);
			normalNode.add(normal[2]);
			
			obj.put("Normal", normalNode);
			
		}
		
		
		if (!obj.isEmpty()){
			return obj;
		}
		return null;
		
	}

	@Override
	public JSONArray serializeArray() {
		// TODO Auto-generated method stub
		return null;
	}
	
}
